stratum/proto/gnmi/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+# Builds gNMI Protobuf library
+#
+# Copyright 2022-2023 Intel Corporation
+# SPDX-License-Identifier: Apache 2.0
+#
+
+cmake_minimum_required(VERSION 3.12)
```

Comment:
The other CMake file has a 3.5 minimum version requirement. Better to keep it consistent with whatever our minimum requirement is.

Reply:
I disagree. The parent CMakeLists.txt file specifies a minimum version of 3.5, which this file inherits. I'm not aware of any features in the file that require a higher version number, which is why I omitted the command. (Bear in mind, the commands in this file were moved here from the parent file.)

We should be _removing_ unnecessary cmake_minimum_required() commands, not adding them!

Reply:
So, this can be removed and it will run with the inherited cmake version 3.5 requirement?

Reply:
Correct.

Reply:
Can we add a `TODO` comment to mark it as potential cleanup in the future? Approving it now, but hopefully this doesnt get lost and forgotten about

Reply:
I've promoted `stratum/CMakeLists.txt` to version 3.12 (the suggested base version for "Modern CMake") and removed the `cmake_minimum_version()` commands from all the subordinate CMakeLists.txt files. This is not a definitive solution, but it gives us a degree of consistency in this subtree, at least for the time being.